app/src/main/java/de/danoeh/antennapod/fragment/PlaybackHistoryFragment.java

```diff
@@ -91,17 +160,49 @@ public void onHistoryUpdated(PlaybackHistoryEvent event) {
     @NonNull
     @Override
     protected List<FeedItem> loadData() {
-        return DBReader.getPlaybackHistory(0, page * EPISODES_PER_PAGE);
+        return DBReader.getPlaybackHistory(0, page * EPISODES_PER_PAGE, getTimeframe());
     }
 
     @NonNull
     @Override
     protected List<FeedItem> loadMoreData(int page) {
-        return DBReader.getPlaybackHistory((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE);
+        return DBReader.getPlaybackHistory((page - 1) * EPISODES_PER_PAGE, EPISODES_PER_PAGE, getTimeframe());
     }
 
     @Override
     protected int loadTotalItemCount() {
-        return (int) DBReader.getPlaybackHistoryLength();
+        return (int) DBReader.getPlaybackHistoryLength(getTimeframe());
+    }
+
+    private long addDayToTimestamp(long timestamp, int day) {
+        // Calculation needed to account for daylight savings time
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(timestamp);
+        calendar.add(Calendar.DAY_OF_MONTH, day);
+        return calendar.getTime().getTime();
+    }
+
+    private void updateFilterUi() {
+        if (isFiltered) {
+            txtvInformation.setVisibility(View.VISIBLE);
+            emptyView.setMessage(R.string.no_all_history_filtered_label);
+        } else {
+            txtvInformation.setVisibility(View.GONE);
+            emptyView.setMessage(R.string.no_history_label);
+        }
+    }
+
+    @NonNull
+    private long[] getTimeframe() {
+        SharedPreferences prefs = getActivity().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        if (!prefs.getBoolean(PREF_FILTERED, false)) {
+            // Adding a day to the end date so that podcasts listened to on the same day are also displayed
+            return new long[]{0, addDayToTimestamp(MaterialDatePicker.todayInUtcMilliseconds(), 1)};
```

Comment:
Can't you just use `System.currentTimeMillis()` here? Or maybe even just `Long.MAX_VALUE`?